control-plane/agents/src/bin/core/main.rs

```diff
@@ -1,6 +1,6 @@
 //! The Core Agent.
 //! todo: document.
-
+#![feature(iter_advance_by)]
```

Comment:
hmm don't think we want to use experimental features..

Reply:
The usage seemed pretty harmless because even if advance_by fails, we will still get the snapshot list just that from beginning index. But if still no consensus on using this, we can replace it with a for loop and do iter.next() within.